packages/experimental/kai/src/pages/SpacePage.tsx

```diff
@@ -46,17 +45,19 @@ const Toolbar = () => {
  * Home page with current space.
  */
 const SpacePage = () => {
-  useDevDataGenerator();
+  useGenerator();
   const { t } = useTranslation('kai');
   const { active } = useFrames();
   const { section } = useParams();
-  const { space, frame } = useFrameState();
-
+  const { space, frame } = useAppRouter();
+  if (!space) {
```

Comment:
Undefined space is handling below already in a way the allows a space to be selected

Reply:
I wasn't keen on there being naked text in the frame
i think after we solve the flickering frame issue, i want to think about a more aesthetic loading and selection ux